haystack/nodes/reader/farm.py

```diff
@@ -1251,6 +1253,73 @@ def _preprocess_batch_queries_and_docs(
 
         return inputs, number_of_docs, single_doc_list
 
+    @staticmethod
+    def _deduplicate_predictions(predictions: List[QAPred], documents: List[Document]) -> List[QAPred]:
```

Comment:
Can we break this function into multiple functions? This will also automatically help us prevent exposing a private method as a `staticmethod`